src/connector.rs

```diff
@@ -58,6 +59,7 @@ impl EthConnectorContract {
             contract: Self::get_contract_data(&EthConnectorStorageId::Contract),
             ft: Self::get_contract_data(&EthConnectorStorageId::FungibleToken),
             paused_mask: Self::get_contract_data(&EthConnectorStorageId::PausedMask),
+            metadata: Self::get_contract_data(&EthConnectorStorageId::FungibleTokenMetadata),
```

Comment:
Thats true. In this case, we would.

Reply:
Possibly a new method the owner can call to add the metadata to their FT?

Reply:
Yes, it should be some migration for existing contracts.

Reply:
But currently, we don't have a migration mechanism. #137 is stopped.

Reply:
We have [this function which is called on migration](https://github.com/aurora-is-near/aurora-engine/blob/develop/src/lib.rs#L182). It could fill in the value we want for existing deployments.

Reply:
Do you mean hardcoded?

Reply:
Yes, hardcoded. Something like

```rust
    pub extern "C" fn state_migration() {
        if connector::read_metadata().is_empty() {
            connector::write_metadata(FungibleTokenMetadata::ETH);
        }
    }
```

Reply:
I think that data should be put from CLI args, not as constants.

Reply:
That might be a more flexible way of doing the migration. Though in my opinion it introduces some complexity without adding much value because in this case we do want to write a constant into the storage.

Reply:
Maybe even having the metadata in storage at all is too much complexity. Taking wrapped NEAR as an example, they do [hardcode the metadata](https://github.com/near/core-contracts/blob/master/w-near/src/lib.rs#L48-L58).

It would save us some trouble if we did the same because we wouldn't need to change the `InitCallArgs` either.

Reply:
I added migration. But let's take a broad discussion on Friday's call.
It'll be easy to change as a hard-coded constant anymore.